doc/user/pmd.rst

```diff
@@ -0,0 +1,205 @@
+   Default value is 80 bytes.
+
+.. index:: [no] packet-tos (1-255)
+.. clicmd:: [no] packet-tos (1-255)
+
+   The type of service value to be written in the appropriate
+   DSCP field of IP packet. Default value is 0xc0 which stands
+   for Inter Network Control value.
+
+.. index:: [no] interval (1-65535)
+.. clicmd:: [no] interval (1-65535)
+
+   Interval between each emission. This value is the interval
+   in milliseconds between each emission of packet. Default
+   interval value is set to 5000 ms.
+
+
+.. index:: [no] timeout (1-65535)
+.. clicmd:: [no] timeout (1-65535)
+
+   Timeout value expressed in milliseconds. This is the time
+   after emission, where one considers that the packet is not
+   received in time. That value can not be greater than
+   interval value. Default value is set to 5000 ms.
+
+
+.. index:: [no] retries down-count (1-255) up-count (1-255)
+.. clicmd:: [no] retries down-count (1-255) up-count (1-255)
+
+   When packets are not received, or not received in time,
+   the pm session is considered as lost. Reversely, the
+   reception of one packet in time makes the pm session go up.
+   However, it is possible to make more flexible the flip-
+   flap algorithm by increasing the retry timers. Those timers
+   permit to consider the session to go up or down, only after
+   a defined amount of received or non received retries.
+
+
+.. index:: [no] shutdown
+.. clicmd:: [no] shutdown
+
+   Enables or disables the peer. When the peer is disabled an
+   'administrative down' message is sent to the remote peer.
+
+.. _pm-static-config:
+
+STATIC PM Configuration
+------------------------
+
+The following commands are available inside the configure or vrf configuration node.
+
+.. index:: ip route [..] GATEWAY pm [..]
+.. clicmd:: ip route [..] GATEWAY pm [..]
+
+.. index:: ipv6 route [..] GATEWAY pm [..]
+.. clicmd:: ipv6 route [..] GATEWAY pm [..]
+
+   Listen for PM events on gateways mentioned in static route created. Every time the
+   gateway is found thanks to nexthop tracking facility, a PM session is created to
+   monitor the route entry. The route entry will be removed dynamically if remote PM
+   session is not operational up.
+
+.. index:: no ip route [..] GATEWAY pm [..]
+.. clicmd:: no ip route [..] GATEWAY pm [..]
+
+.. index:: no ipv6 route [..] GATEWAY pm [..]
+.. clicmd:: no ipv6 route [..] GATEWAY pm [..]
+
+   Removes route entry along with PM session context.
+
+.. _pm-configuration:
+
+Configuration
+=============
+
+Before applying ``pmd`` rules to integrated daemons, we must
+create the corresponding sessions inside the ``pm`` configuration node.
+
+Here is an example of PM configuration:
+
+::
+
+    pm
+    session 192.168.0.2 local-address 192.168.0.1 interface r1-eth0
+     interval 500
+     timeout 500
+     no shutdown
+     !
+    !
+    interface r1-eth0
+     ip address 192.168.0.1/24
+    !
+    ip route 192.168.2.0/24 192.168.0.2 pm vrf r1-cust1
```

Comment:
if we are using PM to monitor nexthop of a static route then why not use BFD? What are the advantages of ICMP here? If packet size is the only deal here then https://tools.ietf.org/html/draft-ietf-bfd-large-packets-02 defines the same.

Reply:
Many services may not support BFD, e.g. load-balancers.